clippy_lints/src/doc.rs

```diff
@@ -417,8 +417,9 @@ fn check_doc<'a, Events: Iterator<Item = (pulldown_cmark::Event<'a>, Range<usize
                     // text "http://example.com" by pulldown-cmark
                     continue;
                 }
-                headers.safety |= in_heading && text.trim() == "Safety";
-                headers.errors |= in_heading && text.trim() == "Errors";
+                let trimmed = text.trim();
+                headers.safety |= has_header(trimmed, in_heading, "# Safety");
```

Comment:
We should check that we are not `in_code`, otherwise this passes the test:

```rust
/// This is not # Safety section.
/// ```
///    # Safety
/// ```
pub unsafe fn ceci_n_est_pas_un_titre() {
    unimplemented!();
}
```

If we add the check (and the missing space in the test to make it 4) the header is not recognised.